### PR TITLE
add ConfigurationManager to Version.Details

### PR DESCRIPTION
**Pull request by oleksandr-didyk**
### Context

Contributes to https://github.com/dotnet/source-build/issues/3043

Declaring the `System.Configuration.ConfigurationManager` dependency in `Version.Details.xml` will allow source-build to replace the currently used `7.0.0` version with the `n-1` version coming from previously source-built artifacts in the product / VMR build.

Without this change, once repo PvP is enabled, the source-build of `msbuild` will fail in the product build.

### Changes Made

  - added an entry for `System.Configuration.ConfigurationManager: 7.0.0` to `Version.Details.xml`

### CODE REVIEW

**Review by JanKrivanek — APPROVED** [review]

**Comment by oleksandr-didyk** [comment]
@rainersigwald soft ping - would be great if you could take a look at this small PR for source-build. Thank you for your time!

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -36,5 +36,11 @@
       <Uri>https://github.com/dotnet/arcade</Uri>
       <Sha>eeb78d042d8f2bef222c83790fa253c5a22675ce</Sha>
     </Dependency>
+    <!-- Necessary for source-build. This allows the package to be retrieved from previously-source-built artifacts
+         and flow in as dependencies of the packages produced by msbuild. -->
+    <Dependency Name="System.Configuration.ConfigurationManager" Version="7.0.0">
```
Comment (rainersigwald) [inline eng/Version.Details.xml:41]: It seems like this should be considered a product dependency, right?

**Review by rainersigwald — COMMENTED** [review]
We have several other runtime dependencies that feel like they might be similar. Should they be added here? If not, how can we tell?

https://github.com/dotnet/msbuild/blob/f3dcaf7e54b794930692221dd9b978ad0582366c/eng/Versions.props#L29-L43

File: eng/Version.Details.xml (hunk 1 above)
Comment (oleksandr-didyk) [reply eng/Version.Details.xml:41]: Yes, I think you are right. Thanks for noticing, will move

**Comment by oleksandr-didyk** [comment]
> We have several other runtime dependencies that feel like they might be similar. Should they be added here? If not, how can we tell?
> 
> https://github.com/dotnet/msbuild/blob/f3dcaf7e54b794930692221dd9b978ad0582366c/eng/Versions.props#L29-L43

In most cases addition of these kind of entries to `Version.Details.xml` is related to the dependency in question being loaded in during the build and source-build having a reference package for it, which causes a build failure. These can be unearth by simply building the repo / product.

Currently only `System.Configuration.ConfigurationManager` is causing problems. With the work for the issue mentioned in the PRs description complete, similar issues would be caught either in PRs to the repo or to `installer`. In .NET 9 timeframe, with the development of VMR, these issues would be caught in the repo PR only.

**Review by rainersigwald — APPROVED** [review]

**Pull request merged**